spyder/app/mainwindow.py

```diff
@@ -2672,123 +2673,11 @@ def add_to_toolbar(self, toolbar, widget):
-            <a href="https://winpython.github.io/">WinPython</a>
-            also contribute to this plan.
-            </p>
-            <p>
-            Python {python_ver} {bitness}-bit | Qt {qt_ver} |
-            {qt_api} {qt_api_ver} | {os_name} {os_ver}
-            </p>
-            <p><small>Certain source files under other compatible permissive
-            licenses and/or originally by other authors.
-            Spyder 3 theme icons derived from
-            <a href="https://fontawesome.com/">Font Awesome</a> 4.7
-            (&copy; 2016 David Gandy; SIL OFL 1.1) and
-            <a href="http://materialdesignicons.com/">Material Design</a>
-            (&copy; 2014 Austin Andrews; SIL OFL 1.1).
-            Most Spyder 2 theme icons sourced from the
-            <a href="https://www.everaldo.com">Crystal Project iconset</a>
-            (&copy; 2006-2007 Everaldo Coelho; LGPL 2.1+).
-            Other icons from
-            <a href="http://p.yusukekamiyamane.com/">Yusuke Kamiyamane</a>
-            (&copy; 2013 Yusuke Kamiyamane; CC-BY 3.0),
-            the <a href="http://www.famfamfam.com/lab/icons/silk/">FamFamFam
-            Silk icon set</a> 1.3 (&copy; 2006 Mark James; CC-BY 2.5), and
-            the <a href="https://www.kde.org/">KDE Oxygen icons</a>
-            (&copy; 2007 KDE Artists; LGPL 3.0+).</small>
-            </p>
-            <p>
-            See the <a href="{github_url}/blob/master/NOTICE.txt">NOTICE</a>
-            file for full legal information.
-            </p>
-            </div>
-            """.format(
-                spyder_ver=versions['spyder'],
-                revision=revlink,
-                website_url=__website_url__,
-                github_url=__project_url__,
-                trouble_url=__trouble_url__,
-                forum_url=__forum_url__,
-                python_ver=versions['python'],
-                bitness=versions['bitness'],
-                qt_ver=versions['qt'],
-                qt_api=versions['qt_api'],
-                qt_api_ver=versions['qt_api_ver'],
-                os_name=versions['system'],
-                os_ver=versions['release'],
-                font_family=font_family,
-                font_size=font_size,
-            )
-        )
-        msgBox.setWindowTitle(_("About %s") % "Spyder")
-        msgBox.setStandardButtons(QMessageBox.Ok)
-
-        from spyder.config.gui import is_dark_interface
-        if is_dark_interface():
-            icon_filename = "spyder.svg"
-        else:
-            icon_filename = "spyder_dark.svg"
-        app_icon = QIcon(get_image_path(icon_filename))
-        msgBox.setIconPixmap(app_icon.pixmap(QSize(64, 64)))
-
-        msgBox.setTextInteractionFlags(
-            Qt.LinksAccessibleByMouse | Qt.TextSelectableByMouse)
-        msgBox.exec_()
+    def show_about(self):
+        """Show About Spyder dialog box"""
+        from spyder.widgets.about import AboutDialog

```

Comment:
I did, thanks. It was both fun and educational, and I would be be happy to do this again. Will do, but under `Local gui imports` it says:
```
# NOTE: Move (if possible) import's of widgets and plugins exactly where they
# are needed in MainWindow to speed up perceived startup time (i.e. the time
# from clicking the Spyder icon to showing the splash screen).
```
so I followed the structure of `show_dependencies()` in line 2685 and included it in the `show_about()` method. I'm happy to move it, but I'm not sure why it should be moved.

Reply:
You are completely right! I forgot about this, please revert this change and put the import inside the function `show_about`

Reply:
Done. Rather than revert, I have made the change. I realised I had added an import for QPushButton which was no longer needed after the AboutDialog was moved outside of mainwindow.py so that needed to be removed.